uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -48,26 +51,32 @@ define([
       region: null,
       contacts_with_leaves: false
     };
-    vm.months = [
-      {label: 'January', loading: false},
-      {label: 'February', loading: false},
-      {label: 'March', loading: false},
-      {label: 'April', loading: false},
-      {label: 'May', loading: false},
-      {label: 'June', loading: false},
-      {label: 'July', loading: false},
-      {label: 'August', loading: false},
-      {label: 'September', loading: false},
-      {label: 'October', loading: false},
-      {label: 'November', loading: false},
-      {label: 'December', loading: false}
-    ];
     vm.loading = {
       calendar: false,
       page: false
     };
-    vm.selectedMonths = [];
-    vm.selectedPeriod = null;
+    vm.monthLabels = ['January', 'February', 'March', 'April', 'May', 'June',
+      'July', 'August', 'September', 'October', 'November', 'December'];
+
+    /**
+     * Fetch all the months from the current period and
+     * save it in vm.months
+     */
+    function fetchMonthsFromPeriod () {
+      var months = [],
+        startDate = moment(vm.selectedPeriod.start_date),
+        endDate = moment(vm.selectedPeriod.end_date);
+
+      while (startDate.isBefore(endDate)) {
```

Comment:
gotta say, i find it really annoying that i can't find an equivalent functional programming to this!